File: Array_String/1-unique/Cpp/unique.cpp
#include <iostream>
#include <string>

using namespace std;

bool isUniqueChars(string str) {
    if (str.length() > 128) return false; // Assume ASCII character set

    bool char_set[128] = { false }; // Initialize all elements to false
    for (int i = 0; i < str.length(); i++) {
        int val = str[i];
        if (char_set[val]) return false; // Already found this character in the string

        char_set[val] = true;
    }
    return true;
}

int main() {
    string inputString;
    cout << "Enter a string: ";
    getline(cin, inputString);

    if (isUniqueChars(inputString)) {
        cout << "All characters in the string are unique." << endl;
    } else {
        cout << "There are duplicate characters in the string." << endl;
    }

    return 0;
}

File: Array_String/1-unique/Python/unique.py
def is_unique_chars(string):
    if len(string) > 128:
        return False

    char_set = [False for _ in range(128)]
    for char in string:
        val = ord(char)
        if char_set[val]:
            return False
        char_set[val] = True
    return True

input_string = input("Enter a string: ")

if is_unique_chars(input_string):
    print("All characters in the string are unique.")
else:
    print("There are duplicate characters in the string.")
